### PR TITLE
moving filterings to one function (checkFilters) and use it in case of redirect too

### PR DESCRIPTION
**Pull request by joelazar**
The url filters and disallowed url filters were not applied to the requests, which were redirected. I fixed it with this pr + added a test for it too. Let me know what you think about it.

### CODE REVIEW

File: colly.go [hunk 1]
```diff
@@ -1285,10 +1292,9 @@ func (c *Collector) Clone() *Collector {
 
 func (c *Collector) checkRedirectFunc() func(req *http.Request, via []*http.Request) error {
 	return func(req *http.Request, via []*http.Request) error {
-		if !c.isDomainAllowed(req.URL.Hostname()) {
-			return fmt.Errorf("Not following redirect to %s because its not in AllowedDomains", req.URL.Host)
+		if err := c.checkFilters(req.URL.String(), req.URL.Hostname()); err != nil {
+			return err
```
Comment (WGH-) [inline colly.go:1296]: I'd preserve "not following redirect to" part, as it's handy to know that it's redirect that failed the check. E.g. `fmt.Errorf("Not following redirect to %q: %w", req.URL, err)`.

(with `%w`, you can still write the test properly, except you'll need to use `errors.Is`)

File: colly_test.go [hunk 2]
```diff
@@ -101,7 +103,6 @@ func newTestServer() *httptest.Server {
 
 	mux.Handle("/redirect", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		http.Redirect(w, r, "/redirected/", http.StatusSeeOther)
-
```
Comment (WGH-) [inline colly_test.go:104]: I agree that all of these formatting changes are good, but let's keep only the relevant changes in the patch.

**Review by WGH- — CHANGES_REQUESTED** [review]

File: colly.go (hunk 1 above)
Comment (joelazar) [reply colly.go]: Thanks, corrected now.

File: colly_test.go (hunk 2 above)
Comment (joelazar) [reply colly_test.go]: Yep, sorry, I totally forgot about my auto-format settings when I did this pr. I'm using [gofumpt](https://github.com/mvdan/gofumpt) BTW.

**Review by WGH- — APPROVED** [review]

**Pull request merged**